vald/evm/evm.go

```diff
@@ -460,7 +467,7 @@ func (mgr Mgr) GetTxReceiptIfFinalized(chain nexus.ChainName, txID common.Hash,
 	}
 
 	if txReceipt.Status != geth.ReceiptStatusSuccessful {
-		return nil, errors.With(fmt.Errorf("transaction %s failed", txID.Hex()), keyvals...)
+		return nil, nil
```

Comment:
Because the function is only supposed to return an error when it cannot determine if a tx is successful and finalized. Returning an error here is a misuse if you look at the usage of it. The following refactor will make it nicer but for the context of bug fixing I'd rather keep things consistent.